storm-core/src/jvm/org/apache/storm/cluster/IStormClusterState.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.cluster;
+
+import org.apache.storm.generated.*;
+import org.apache.storm.nimbus.NimbusInfo;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Map;
+
+public interface IStormClusterState {
+    public List<String> assignments(Runnable callback);
+
+    public Assignment assignmentInfo(String stormId, Runnable callback);
+
+    public Map assignmentInfoWithVersion(String stormId, Runnable callback);
+
+    public Integer assignmentVersion(String stormId, Runnable callback) throws Exception;
+
+    public List<String> blobstoreInfo(String blobKey);
+
+    public List nimbuses();
+
+    public void addNimbusHost(String nimbusId, NimbusSummary nimbusSummary);
+
+    public List<String> activeStorms();
+
+    public StormBase stormBase(String stormId, Runnable callback);
+
+    public ClusterWorkerHeartbeat getWorkerHeartbeat(String stormId, String node, Long port);
+
+    public List<ProfileRequest> getWorkerProfileRequests(String stormId, NodeInfo nodeInfo, boolean isThrift);
+
+    public List<ProfileRequest> getTopologyProfileRequests(String stormId, boolean isThrift);
```

Comment:
The isThrift is ignored by all of the implementations.  We should just remove it if we are not going to support it.